File: src/main/java/poc/interaction/PersonQueryResolver.java
package poc.interaction;

import com.coxautodev.graphql.tools.GraphQLQueryResolver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import poc.model.Person;
import poc.service.PersonService;

import java.util.List;
import java.util.Optional;

@Component
public class PersonQueryResolver implements GraphQLQueryResolver {
    private final PersonService service;

    @Autowired
    public PersonQueryResolver(PersonService service) {
        this.service = service;
    }

    public List<Person> getAll() {
        return service.getAllPeople();
    }

    public Optional<Person> getOne(int id) {
        return service.getPersonById(id);
    }
}

File: src/main/kotlin/poc/resolver/PersonQueryResolver.kt
package poc.resolver

import com.coxautodev.graphql.tools.GraphQLQueryResolver
import graphql.schema.DataFetchingEnvironment
import graphql.servlet.GraphQLContext
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import poc.loader.CustomGraphQlContext.Companion.PERSON_DATA_LOADER
import poc.model.Person
import poc.repository.PersonRepository

@Component
class PersonQueryResolver(@Autowired private val repository: PersonRepository) : GraphQLQueryResolver {
    fun getAll() = repository.findAll()

    fun getOne(id: Int) = repository.findById(id)

    fun getByDataLoader(ids: List<Int>, dataFetchingEnvironment: DataFetchingEnvironment): List<Person> {
        val context = dataFetchingEnvironment.getContext<GraphQLContext>()
        val personDataLoader = context.dataLoaderRegistry.get().getDataLoader<Int, Person>(PERSON_DATA_LOADER)

        val promise = personDataLoader.loadMany(ids)
        personDataLoader.dispatch()

        return promise.get()
    }

}
